legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java

```diff
@@ -282,4 +314,22 @@ public static Selection getRowsBasedOnLatestTimestamp(Dataset dataset, String fi
 
         return Selection.builder().source(dataset.datasetReference()).condition(condition).addFields(countFunction).build();
     }
+
+    public static void validateOptimizationFilters(List<OptimizationFilter> optimizationFilters, Dataset dataset)
+    {
+        for (OptimizationFilter filter: optimizationFilters)
+        {
+            Field filterField = dataset.schema().fields().stream()
+                    .filter(field -> field.name().equalsIgnoreCase(filter.fieldName()))
+                    .findFirst().orElseThrow(() -> new IllegalStateException(String.format("Optimization filter [%s] not found in Staging Schema", filter.fieldName())));
```

Comment:
Should we validate the existence of the field in main (or both main and stage)? As we eventually are going to use this column from main in the condition (we do have an edge case where stage can have fewer columns than main)